llvm/docs/ProgrammersManual.rst

```diff
@@ -164,20 +164,20 @@ rarely have to include this file directly).
   efficient to use the ``InstVisitor`` class to dispatch over the instruction
   type directly.
 
-``isa_and_nonnull<>``:
-  The ``isa_and_nonnull<>`` operator works just like the ``isa<>`` operator,
+``isa_and_present<>``:
+  The ``isa_and_present<>`` operator works just like the ``isa<>`` operator,
```

Comment:
I can confirm the other two are deprecated but not this one. Where specifically are you getting that from?

Reply:
> I can confirm the other two are deprecated but not this one. Where specifically are you getting that from?

For consistency that suggesting using function templates whose name ends with "\_present".

Reply:
I see, isa_and_nonnull just calls isa_and_present.